### DISCUSSION TITLE
What is the rule about apparency of invoke_xxx  calling function from Emscripten.js function to WASM function

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#20768] [Q&A]
When I look into the chrome debug call stuck I see the Emscripten.js `invoke_xxx` function in the middle of two c++ functions.
It is not clear when Emscripten.js  use that `invoke_xxx` function and who can I force pure c++ without  go out to JS.

**Comment by sbc100** [comment]
The `invoke_xxx` are used to implement C++ exception handling and setjmp/longjmp because Wasm historically doesn't have any control flow structure like try/catch.

If you want to avoid those you can build without exceptions and setjmp/longjmp support (`-sSUPPORT_LONGJMP=0`).   Or you can try building with the new wasm exception handling support `-fwasm-exceptions`.
